README.md

```diff
@@ -1,3 +1,267 @@
+    public DefinitionHeader header() {
+        return new DefinitionHeader("Foo LZ", "Description of Foo LZ");
+    }
+
+    @Override
+    public List<DefinitionVersion> availableVersions() {
+        return List.of(DefinitionVersion.V1);
+    }
+
+    @Override
+    public LandingZoneDefinable create(DefinitionVersion version) {
+        if (version.equals(DefinitionVersion.V1)) {
+            return new FooLZDefinitionV1(azureResourceManager, relayManager);
+        }
+        throw new RuntimeException("Invalid Version");
+    }
+}
+```
+
+An inner class in the factory class is a good convention for implementing a Landing Zone Definition.
+
+```java
+public class FooLZFactory extends ArmClientsDefinitionFactory {
+    ...
+
+    class FooLZDefinitionV1 extends LandingZoneDefinition {
+
+        protected FooLZDefinitionV1(
+                AzureResourceManager azureResourceManager, RelayManager relayManager) {
+            super(azureResourceManager, relayManager);
+        }
+
+        @Override
+        public Deployable definition(DefinitionContext definitionContext) {
+            var storage =
+                    azureResourceManager
+                            .storageAccounts()
+                            .define(definitionContext.resourceNameGenerator().nextName(20))
+                            .withRegion(Region.US_EAST2)
+                            .withExistingResourceGroup(definitionContext.resourceGroup());
+
+            var vNet =
+                    azureResourceManager
+                            .networks()
+                            .define(definitionContext.resourceNameGenerator().nextName(20))
+                            .withRegion(Region.US_EAST2)
+                            .withExistingResourceGroup(definitionContext.resourceGroup())
+                            .withAddressSpace("10.0.0.0/28")
+                            .withSubnet("compute", "10.0.0.0/29")
+                            .withSubnet("storage", "10.0.0.8/29");
+
+            return definitionContext
+                    .deployment()
+                    .withResourceWithPurpose(storage, ResourcePurpose.SHARED_RESOURCE)
+                    .withVNetWithPurpose(vNet, "compute", SubnetResourcePurpose.WORKSPACE_COMPUTE_SUBNET)
+                    .withVNetWithPurpose(vNet, "storage", SubnetResourcePurpose.WORKSPACE_STORAGE_SUBNET);
+        }
+    }
+```
+
+Resources are defined using the standard Azure Java SDK but with the following constraints to consider:
+
+- The purpose of the resources must be indicated in the deployment.
+- Resources not included in the deployment won't be created.
+- The `create()` method in the resource definition must not be called.
+- The resource definition must have the required configuration for a creation before it can be added to the deployment.
+
+### Receiving Parameters
+
+The `DefinitionContext` contains the property `Map<String, String> parameters`, which can be used to receive parameter
```

Comment:
Should we add a list of supported parameters here? It is pretty unclear which parameters can be used? My understanding that these parameters are applicable to different Azure resources. Should we list them all or add link to Azure documentation?

Reply:
I think it is a good idea to document the parameters for each landing zone. I will add section for that.